app/src/pages/inside/dashboardItemPage/modals/common/widgetControls/controls/attributesFieldArrayControl/attributesFieldArrayControl.jsx

```diff
@@ -0,0 +1,159 @@
+import styles from './attributesFieldArrayControl.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  attributeKeyFieldLabel: {
+    id: 'AttributesFieldArrayControl.attributeKeyFieldLabel',
+    defaultMessage: 'Level {number}',
+  },
+  attributeKeyFieldLabel_0: {
+    id: 'AttributesFieldArrayControl.attributeKeyFieldLabelOverview',
+    defaultMessage: 'Level 1 (overview)',
+  },
+  attributeKeyFieldLabel_1: {
+    id: 'AttributesFieldArrayControl.attributeKeyFieldLabelDetailedView',
+    defaultMessage: 'Level 2 (detailed view)',
+  },
+  attributeKeyFieldPlaceholder: {
+    id: 'AttributesFieldArrayControl.attributeKeyFieldPlaceholder',
+    defaultMessage: 'Enter an attribute key',
+  },
+  addOneMoreLevel: {
+    id: 'AttributesFieldArrayControl.addOneMoreLevel',
+    defaultMessage: '+ Add one more level',
+  },
+});
+
+@connect((state) => ({
+  launchAttributeKeysSearch: URLS.launchAttributeKeysSearch(activeProjectSelector(state)),
+  itemAttributeKeysAllSearch: URLS.itemAttributeKeysAllSearch(activeProjectSelector(state)),
+}))
+@injectIntl
+export class AttributesFieldArrayControl extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    fields: PropTypes.object.isRequired,
+    launchAttributeKeysSearch: PropTypes.string.isRequired,
+    itemAttributeKeysAllSearch: PropTypes.string.isRequired,
+    fieldValidator: PropTypes.func.isRequired,
+    maxAttributesAmount: PropTypes.number.isRequired,
+    widgetType: PropTypes.string.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    if (!props.fields.length) {
+      props.fields.push('');
+    }
+  }
+
+  getAttributes = () => this.props.fields.getAll() || [];
+
+  makeAttributes = (items) => {
+    const filteredItems = items.filter((item) => !this.getAttributes().includes(item));
+
+    return filteredItems ? filteredItems.map((item) => ({ value: item, label: item })) : null;
+  };
+
+  isOptionUnique = ({ option }) => !this.getAttributes().includes(option.value);
+
+  formatAttributes = (attribute) => (attribute ? { value: attribute, label: attribute } : null);
+
+  parseAttributes = (attribute) => {
+    if (attribute === null) return null;
+    if (attribute && attribute.value) return attribute.value;
+
+    return undefined;
+  };
+
+  render() {
+    const {
+      intl: { formatMessage },
+      fields,
+      launchAttributeKeysSearch,
+      itemAttributeKeysAllSearch,
+      fieldValidator,
+      maxAttributesAmount,
+      widgetType,
+    } = this.props;
+    const canAddNewItems = fields.length < maxAttributesAmount;
+
+    return (
+      <div className={cx('attribute-list')}>
+        <ScrollWrapper>
+          {fields.map((item, index) => {
+            const isFirstItem = index === 0;
+            return (
+              <ModalField
+                // eslint-disable-next-line
+                key={item}
+                label={
+                  widgetType === CUMULATIVE_TREND
```

Comment:
I think this component shouldn't know anything about widget type. Please manage this condition by passing corresponding props.